src/App.tsx

```diff
@@ -1,61 +1,92 @@
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './components/types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const updatedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(updatedTodos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<number>(0);
+  const [error, setError] = useState(false);
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(+event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setError(false);
+    if (!title || !selectedUser) {
+      setError(true);
+
+      return;
+    }
+
+    const maxId = Math.max(...todos.map(todo => todo.id));
+    const newTodo = {
+      id: maxId + 1,
+      title: title,
+      completed: false,
+      userId: selectedUser,
+      user: usersFromServer.find(u => u.id === selectedUser),
+    };
+
+    setTodos([...todos, newTodo]);
```

Comment:
always when using prev state:
```suggestion
    setTodos(prevTodos => [...prevTodos, newTodo]);
```